core/src/player.rs

```diff
@@ -679,8 +698,40 @@ impl Player {
                     ContextMenuCallback::Forward => Self::forward_root_movie(context),
                     ContextMenuCallback::Back => Self::back_root_movie(context),
                     ContextMenuCallback::Rewind => Self::rewind_root_movie(context),
-                    ContextMenuCallback::Avm2 { .. } => {
-                        // TODO: Send menuItemSelect event
+                    ContextMenuCallback::Avm2 { item } => {
+                        // TODO: This should use the pointed display object (see comment on line 614)
+                        let root_dobj = context.stage.root_clip();
+
+                        if let Some(root_dobj) = root_dobj {
+                            let menu_item = *item;
+                            let mut activation = Avm2Activation::from_nothing(context.reborrow());
+
+                            let menu_obj = root_dobj
+                                .as_interactive()
+                                .map(|iobj| iobj.context_menu())
+                                .and_then(|v| v.as_object());
+
+                            if menu_obj.is_some() {
+                                // TODO: contextMenuOwner and mouseTarget might not be the same (see above comment)
+                                let menu_evt = activation
```

Comment:
all of this should have been extracted to its own method, just like `run_context_menu_custom_callback` for AVM1